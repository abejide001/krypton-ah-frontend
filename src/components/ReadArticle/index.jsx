import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Image, Container, Header, Comment, Divider } from 'semantic-ui-react';
import './styles/ReadArticle.scss';
import calendar from './assets/calendar.svg';
import time from './assets/time.svg';
import fbk from './assets/fbk.svg';
import linkd from './assets/linkdn.svg';
import gmail from './assets/gmail.svg';
import twitter from './assets/twitter.svg';

export default class ReadArticle extends Component {
  getArticleContent = articleContent => (
    <Container textAlign="justified">
      <div className="icon-bar">
        <a href="#" className="facebook">
          <Image src={fbk} />
        </a>
        <a href="#" className="twitter">
          <Image src={twitter} />
        </a>
        <a href="#" className="google">
          <Image src={gmail} />
        </a>
        <a href="#" className="linkedin">
          <Image src={linkd} />
        </a>
      </div>
      <div className="articleContent">
        <p>{articleContent}</p>
        <p>{articleContent}</p>
      </div>
    </Container>
  );

  getArticleInfo = (articleAuthor, articleDate, articleReadTime) => (
    <Comment className="infoContainer">
      <Comment.Content>
        <Comment.Author className="articleAuthor">{articleAuthor}</Comment.Author>
        <Comment.Text className="articleDate">
          <Image src={calendar} />
          <span>{articleDate}</span>
        </Comment.Text>
        <Comment.Text className="articleReadTime">
          <Image src={time} />
          <span>{articleReadTime}</span>
        </Comment.Text>
      </Comment.Content>
    </Comment>
  );

  render() {
    const {
      articleTitle,
      articleAuthor,
      articleDate,
      articleReadTime,
      articleContent,
      featuredImage,
      author
    } = this.props;
    return (
      <Container className="readArticleContainer">
        <Image src={featuredImage} size="massive" />
        <Header as="h1" className="articleTitle">
          {articleTitle}
        </Header>
        <Comment.Group className="articleInfo">
          <Image avatar as="a" src={author} />
          {this.getArticleInfo(articleAuthor, articleDate, articleReadTime)}
        </Comment.Group>
        <Divider className="articleDivider" />
        {this.getArticleContent(articleContent)}
        <Divider className="articleDivider" />
      </Container>
    );
  }
}
ReadArticle.propTypes = {
  articleTitle: PropTypes.string.isRequired,
  articleAuthor: PropTypes.string.isRequired,
  articleDate: PropTypes.string.isRequired,
  articleReadTime: PropTypes.string.isRequired,
  articleContent: PropTypes.string.isRequired,
  featuredImage: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired
};
