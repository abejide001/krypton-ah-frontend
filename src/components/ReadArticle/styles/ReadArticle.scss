@import '../../../styles/partials/colors';
@import '../../../styles/partials/fonts';

.ui.header {
  font-family: 'Roboto'}

.ui.header.articleTitle {
  font-size: 2.25rem;
  color: $black !important;
  text-transform: uppercase;
  padding-left: 3%; 
}


.ui.comments .comment .author.articleAuthor {
  font-size: 0.75rem;
  font-weight: 700;
  line-height: 2;
  text-transform: uppercase;
}

.ui.comments .comment .text.articleDate {
  font-size: 0.75rem;
  font-weight: 100;
  font-style: italic;
  margin-bottom: 0;
  padding-left: 2%; }
  .ui.comments .comment .text.articleDate span {
    padding-left: 5%; }

.ui.comments .comment .text.articleReadTime {
  font-size: 0.5625rem;
  font-weight: 100;
  margin-top: 0;
  padding-left: 2%; }
  .ui.comments .comment .text.articleReadTime span {
    padding-left: 5%; }

.ui.comments .comment.infoContainer {
  margin: -1% 1%;
  padding: 0%;
  width: 20%; }

img.ui.image {
  display: inline; }

.ui.comments.articleInfo {
  padding-left: 3%;
  display: flex; }

.ui.divider.articleDivider {
  margin: 0.5rem 0 0 2.7rem;
  width: 90%; }

.articleContent {
  margin: 2% 7% 10% 6%; }

.articleContent p {
  font-size: 0.9rem;
  line-height: 2rem;
  font-weight: 400;
  background-color: transparent;
color: black
}

.icon-bar {
  position: fixed;
  top: 50%;
  left: 3%; }
  .icon-bar a {
    display: block;
    padding: 15%;
    text-align: center;
    font-size: 1.25rem; }

.readArticleContainer {
  margin-top: 0.5%;
  padding: 0 5%; }

.ui.comments .comment > .content{
  padding-top: 0;
  width: 100%
}

@media only screen and (max-width: 991px) {
  .ui.container.readArticleContainer {
    width: 100vw !important;
    padding: 0;
    margin: 0 !important; }
  .ui.justified.container {
    position: relative;
    width: 100%; }
  .icon-bar {
    width: 50%;
    position: absolute;
    top: unset;
    bottom: -2rem;
    left: 5%; }
    .icon-bar a {
      display: inline;
      padding: 5%; } }
