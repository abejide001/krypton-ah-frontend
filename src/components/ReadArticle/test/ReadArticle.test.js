import React from 'react';
import { shallow } from 'enzyme';
import ReadArticle from '../index';
import { author,
  featuredImage,
  ARTICLE_TITLE,
  ARTICLE_AUTHOR,
  ARTICLE_DATE,
  ARTICLE_READ_TIME,
  ARTICLE_CONTENT } from '../../../constants';

describe('ReadArticle', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <ReadArticle
        articleTitle={ARTICLE_TITLE}
        articleAuthor={ARTICLE_AUTHOR}
        articleDate={ARTICLE_DATE}
        articleReadTime={ARTICLE_READ_TIME}
        articleContent={ARTICLE_CONTENT}
        featuredImage={featuredImage}
        author={author}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
