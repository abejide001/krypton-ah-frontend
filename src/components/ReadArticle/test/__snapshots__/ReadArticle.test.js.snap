// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read article component should matches the snapshot 1`] = `
<ReadArticle
  articleAuthor={Object {}}
  content=""
  createdAt=""
  featuredImageUrl=""
  getArticle={[Function]}
  readTime={0}
  retrievedArticle={
    Object {
      "response": Object {
        "articleAuthor": Object {
          "firstname": "Michael",
          "lastname": "James",
          "userprofile": null,
        },
        "content": "this is a good content wooooooooooooooooooot tooooooooooooooooooooooooot",
        "createdAt": "2019-02-20T16:53:37.716Z",
        "featuredImageUrl": "https://res.cloudinary.com/ah-krypton/image/upload/v1550010452/widget.png",
        "readTime": 1,
        "title": "this is a valid articles 2",
      },
    }
  }
  selectedArticle="65719288-0395-445e-b587-2b98b70bdec9"
  title=""
>
  <Container
    className="readArticleContainer"
  >
    <div
      className="ui container readArticleContainer"
    >
      <Image
        as="img"
        size="massive"
        src="https://res.cloudinary.com/ah-krypton/image/upload/v1550010452/widget.png"
        ui={true}
      >
        <img
          className="ui massive image"
          src="https://res.cloudinary.com/ah-krypton/image/upload/v1550010452/widget.png"
        />
      </Image>
      <Header
        as="h1"
        className="articleTitle"
      >
        <h1
          className="ui header articleTitle"
        >
          this is a valid articles 2
        </h1>
      </Header>
      <CommentGroup
        className="articleInfo"
      >
        <div
          className="ui comments articleInfo"
        >
          <Image
            as="a"
            avatar={true}
            src="test-file-stub"
            ui={true}
          >
            <a
              className="ui avatar image"
            >
              <img
                src="test-file-stub"
              />
            </a>
          </Image>
          <Comment
            className="infoContainer"
          >
            <div
              className="comment infoContainer"
            >
              <CommentContent>
                <div
                  className="content"
                >
                  <CommentAuthor
                    className="articleAuthor"
                  >
                    <div
                      className="author articleAuthor"
                    >
                      Michael James
                    </div>
                  </CommentAuthor>
                  <CommentText
                    className="articleDate"
                  >
                    <div
                      className="articleDate text"
                    >
                      <Image
                        as="img"
                        src="test-file-stub"
                        ui={true}
                      >
                        <img
                          className="ui image"
                          src="test-file-stub"
                        />
                      </Image>
                      <span>
                        Wed Feb 20 2019 
                      </span>
                    </div>
                  </CommentText>
                  <CommentText
                    className="articleReadTime"
                  >
                    <div
                      className="articleReadTime text"
                    >
                      <Image
                        as="img"
                        src="test-file-stub"
                        ui={true}
                      >
                        <img
                          className="ui image"
                          src="test-file-stub"
                        />
                      </Image>
                      <span>
                        1 Mins read
                      </span>
                    </div>
                  </CommentText>
                </div>
              </CommentContent>
            </div>
          </Comment>
        </div>
      </CommentGroup>
      <Divider
        className="articleDivider"
      >
        <div
          className="ui divider articleDivider"
        />
      </Divider>
      <Container
        textAlign="justified"
      >
        <div
          className="ui justified container"
        >
          <div
            className="icon-bar"
          >
            <a
              className="facebook"
              href="#"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              >
                <img
                  className="ui image"
                  src="test-file-stub"
                />
              </Image>
            </a>
            <a
              className="twitter"
              href="#"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              >
                <img
                  className="ui image"
                  src="test-file-stub"
                />
              </Image>
            </a>
            <a
              className="google"
              href="#"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              >
                <img
                  className="ui image"
                  src="test-file-stub"
                />
              </Image>
            </a>
            <a
              className="linkedin"
              href="#"
            >
              <Image
                as="img"
                src="test-file-stub"
                ui={true}
              >
                <img
                  className="ui image"
                  src="test-file-stub"
                />
              </Image>
            </a>
          </div>
          <div
            className="articleContent"
          >
            <p>
              this is a good content wooooooooooooooooooot tooooooooooooooooooooooooot
            </p>
            <p>
              this is a good content wooooooooooooooooooot tooooooooooooooooooooooooot
            </p>
          </div>
        </div>
      </Container>
      <Divider
        className="articleDivider"
      >
        <div
          className="ui divider articleDivider"
        />
      </Divider>
    </div>
  </Container>
</ReadArticle>
`;
